commandAuthoringGuide.md

```diff
@@ -0,0 +1,77 @@
+### Creating a new `nim` subcommand
+
+The Nimbella CLI is built on the [oclif framework](https://oclif.io/docs/introduction.html).  Some familiarity with that framework helps in understanding the following.
+
+The Nimbella CLI includes the [plugins plugin](https://github.com/oclif/plugin-plugins).  So, you have a choice.  You can package additional subcommands in a plugin, or you can submit a PR to modify `nim` itself.  The choice should depend on the general applicability of the subcommand.  If the potential usage community for your new command is narrow, consider using a plugin.
+
+If you do use a plugin,
+
+- you will need to publish it to an `npm` repository or disseminate github coordinates
```

Comment:
Is there a naming convention that one should follow while publishing a plugin to npm?

Reply:
I was thinking that, if we were not taking on the responsibility for either maintaining or publicizing plugins, then it's a "free market" and we would not dictate naming.   

I do think we want to carefully evaluate the names given to built-in commands, but, I don't see an easy way to reduce that to a rule, so we'd just have to look at PRs case by case and decide whether we approve or don't approve the command's name along with everything else.

Reply:
At the very least, names should conform to the COC.

Reply:
I will add a general statement about conforming to the COC to the authoring guide.   There is a similar statement in the CONTRIBUTING document.

Reply:
My concern is that, will nim install any npm package as a plugin? Let's assume that I published a plugin as an npm package with the name `nim-plugin-commander`. To install it, do I need to run `nim plugin install nim-plugin-commander`?

Reply:
Yes, I believe that's what you would type.   I'm not sure what the concern is here.   Let's continue the conversation in slack.   I'm not in a rush to merge this so we can work on addressing all concerns.